### PR TITLE
add support to attribute placeholder

### PR DESCRIPTION
**Pull request by darlanmendonca**
PR to add support to attribute placeholder, following specs in issue https://github.com/minimalist-components/webcomponents/issues/13

@alencar13 get the branch attribute-placeholder to start write specs pls

### CODE REVIEW

**Review by alencar13 — DISMISSED** [review]

**Comment by alencar13** [comment]
@darlanmendonca please, take a look and see if needs to improve.

**Comment by darlanmendonca** [comment]
@alencar13 remove the skips, and ovewrite all to a single spec (expect element have a label with text X)

the style specs will be more easy to write, when we have a demo, I'm working in that now

**Comment by alencar13** [comment]
I changed the last specs to this new spec as new commit but i'm having this problem:

captura de tela de 2017-05-18 18-59-27

I don't know if this problem is because my spec is wrote wrong or because of attribute placeholder is not implemented.

Can you help me?

**Comment by darlanmendonca** [comment]
@alencar13 in your spec, before `expect(element).to.contain('label')` write a log "to get label manually"

```js
console.log(element.querySelector('label'))
expect(element).to.contain('label')
```

if log return `undefined` or `null`, the assertion break because I dont implement placeholder yet.

AND, the assertion error message "flag(...) is null" can be bad.
I suggest to you, do a Issue or PR in chai-dom to improve this message to anything more readable, e.g. "element dont contain 'label"

**Comment by alencar13** [comment]
I inserted the `console.log(element.querySelector('label'))` and really is returning `'null'`.

Now I'll do a issue in chai-dom to improve the message.

File: webcomponents/input/input.webcomponent.spec.js [hunk 1]
```diff
@@ -133,6 +140,29 @@ describe('mn-input (webcomponent)', () => {
...
+      expect(element).to.contain('label').with.text('test')
+    })
+
+    it('should change the placeholder', () => {
+      element.setAttribute('placeholder','test')
+      element.setAttribute('placeholder','test2')
+      expect(element).to.contain('label').with.text('test2')
+    })
+
+    it('should verify the default value when placeholder is not setted', () => {
```
Comment (darlanmendonca) [inline webcomponents/input/input.webcomponent.spec.js]: maybe be better describe it with

"should set empty label if is undefined"

or

"should set empty label if not defined"

File: webcomponents/input/input.webcomponent.spec.js [hunk 2]
```diff
@@ -133,6 +140,29 @@ describe('mn-input (webcomponent)', () => {
...
+      expect(element).to.contain('label').with.text('')
+    })
+
+    it('should set property value when attribute is setted and after removed', () => {
```
Comment (darlanmendonca) [inline webcomponents/input/input.webcomponent.spec.js]: maybe be better write this it like "should empty label when placeholder is removed"

the word "value" looks a little confuse

Comment (alencar13) [reply webcomponents/input/input.webcomponent.spec.js]: It's better! I'll modify :+1:

File: webcomponents/input/input.webcomponent.spec.js (hunk 1 above)
Comment (alencar13) [reply webcomponents/input/input.webcomponent.spec.js]: Good! I'm going to choose to "should set empty label if is undefined".

**Comment by alencar13** [comment]
Adjust done. Is there something more to fix? @darlanmendonca

**Review by alencar13 — DISMISSED** [review]
Approved.

**Comment by darlanmendonca** [comment]
@alencar13 we need more specs, to check 'property' placeholder, e.g.

```js
element.placeholder = 'lero'
```

this change the attribute placeholder, and set the text in child label.
write theses specs to us, pls

**Comment by alencar13** [comment]
@darlanmendonca is broking CI because you need to change the file `input.class.js` in line 98:

`? this.label.textContent = this.getAttribute('placeholder')` to `? this.label.textContent = value`.

Any problem, let me know

**Review by alencar13 — APPROVED** [review]

**Pull request merged**